### PR TITLE
Args not updating bug

### PR DESCRIPTION
**Pull request by robbiepitts**
Fixes https://github.com/glimmerjs/glimmer-component/issues/66.

The component manager is supposed to update `this.args` on a component when they update. This was not happening at the appropriate time, causing computed properties dependent upon `this.args` to not update in time to reflect in the layout.

Args were being updated in the `didUpdate` component manager hook but needed to be updated in the `update` hook. Updating the args in `didUpdate` is too late in the rendering process (`didUpdate` indicates that the component has updated already). The `update` hook is for updating the component ahead of updating the layout so the layout knows what to render.

Also upgrades the @glimmer/application devDependency and removes an unnecessary guard.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,7 +31,7 @@
     "@glimmer/util": "^0.26.2"
   },
   "devDependencies": {
-    "@glimmer/application": "^0.5.0",
+    "@glimmer/application": "^0.7.2",
```
Comment (tomdale) [inline package.json]: Is this upgrade necessary to fix the issue or can it be moved to a separate PR?

File: test/args-test.ts [hunk 2]
```diff
@@ -76,6 +77,106 @@ test('Args smoke test', (assert) => {
...
+        <child-component @firstName={{firstName}} @status={{status}} />
+      </div>`)
+    .template('child-component', '<div>{{name}} {{@status}}</div>')
+    .boot();
+
+  setPropertyDidChange(function() {
+    app.scheduleRerender();
+  });
+
+  assert.equal(app.rootElement.textContent.trim(), 'Tom Dale is dope');
```
Comment (tomdale) [inline test/args-test.ts]: A+++++ tests, would test again

File: test/args-test.ts [hunk 3]
```diff
@@ -76,6 +77,106 @@ test('Args smoke test', (assert) => {
   parent.firstName = "Thomas";
 });
 
+test('Rendering args derivatives', (assert) => {
```
Comment (tomdale) [inline test/args-test.ts]: This is a nitpick but I would prefer test names that are more descriptive, so they are more accessible to future contributors. For example, perhaps instead of "Rendering args derivatives", we could say something like "Tracked properties that depend on `args` re-render correctly". The terse names make it difficult to understand at a glance the difference between the two tests.

File: src/component-manager.ts [hunk 4]
```diff
@@ -133,18 +133,12 @@ export default class ComponentManager implements GlimmerComponentManager<Compone
...
+    bucket.component.args = bucket.namedArgsSnapshot();
   }
 
   didUpdateLayout() {}
 
-  didUpdate(bucket: ComponentStateBucket) {
-    if (!bucket) { return; }
-
-    // TODO: This should be moved to `didUpdate`, but there's currently a
-    // Glimmer bug that causes it not to be called if the layout doesn't update.
```
Comment (tomdale) [inline src/component-manager.ts]: Just to verify, has the Glimmer VM bug that necessitated this been fixed? (Seems like yes.)

Comment (robbiepitts) [inline src/component-manager.ts]: Yes

File: test/args-test.ts (hunk 2 above)
Comment (robbiepitts) [inline test/args-test.ts]: Thanks, Thom.

File: package.json (hunk 1 above)
Comment (robbiepitts) [inline package.json]: Yes, it is necessary for these tests to pass. I could probably do this upgrade in a separate PR still, but it would need to be merged first.

File: test/args-test.ts (hunk 3 above)
Comment (robbiepitts) [inline test/args-test.ts]: 👍

**Comment by robbiepitts** [comment]
@tomdale Updated per review 😃

**Comment by tomdale** [comment]
LGTM! 👍  Thank you for the hard work!
